sled-agent/src/services.rs

```diff
@@ -274,42 +292,168 @@ impl Config {
 }
 
 // The filename of the ledger, within the provided directory.
-const SERVICES_LEDGER_FILENAME: &str = "services.json";
-
-// A wrapper around `ZoneRequest`, which allows it to be serialized
-// to a JSON file.
-#[derive(Clone, serde::Serialize, serde::Deserialize, schemars::JsonSchema)]
-struct AllZoneRequests {
-    generation: Generation,
-    requests: Vec<ZoneRequest>,
-}
+const ZONES_LEDGER_FILENAME: &str = "omicron_zones.json";
+
+/// Combines the Nexus-provided `OmicronZonesConfig` (which describes what Nexus
+/// wants for all of its zones) with the locally-determined configuration for
+/// these zones.
+#[derive(
+    Clone, Debug, serde::Serialize, serde::Deserialize, schemars::JsonSchema,
+)]
+pub struct OmicronZonesConfigLocal {
+    /// version of the Omicron-provided part of the configuration
+    ///
+    /// This version number is outside of Sled Agent's control.  We store
+    /// exactly what we were given and use this number to decide when to
+    /// fail requests to establish an outdated configuration.
+    ///
+    /// You can think of this as a major version number, with `ledger_version`
+    /// being a minor version number.  See `is_newer_than()`.
+    pub omicron_version: Generation,
```

Comment:
Hmm, yeah.  I went back and forth on this a bit.  I think I introduced the term "generation" early in Omicron and it's certainly all over the place now.  I think I got this term from distributed systems work in college maybe?  But during the RFD 367/DNS work I started thinking "how is this different from just a plain version number?"  (When I hear "version number", especially as in "let's put a version number into this data structure", I tend to assume integer, not semver.  This might just reflect different experiences?)  Anyway, I googled around and had trouble finding even a _definition_ of "generation number" in the sense that we use it, which made me feel even more like this choice of term was more obscure than necessary.

Just now I asked ChatGPT (ha!), and its summary matches what I thought and certainly fits this use case, so I probably shouldn't have second-guessed it:
![Screenshot 2023-11-29 at 1 00 28 PM](https://github.com/oxidecomputer/omicron/assets/532688/f59ab358-269e-44ee-b90a-076d812d507f)

I'll just go ahead and change it back.

Reply:
Fixed in 4274e9d26a05af7c49328fdffff0b9997fcb4029